controllers/blogController.js

```diff
@@ -0,0 +1,220 @@
+
+const Blog = require('../models/blogModel');
+const BlogDetails = require('../models/blogDetailModel');
+
+const getAllBlogs = async (req, res) => {
+
+    let offset = req.query.offset;
+    let limit = req.query.limit;
+
+    console.log(limit);
+
+    try {
+        const blogs = await Blog.find().skip(offset).limit(limit).populate('author', 'name email -_id');
+        res.status(200).json(
+            blogs
+        )
+    } catch (err) {
+        console.log(err);
+        res.status(500).json({
+            msg: "something went wrong, try again!!!",
+        })
+    }
+}
+
+const getPopularBlogs = async (req, res) => {
+
+    res.status(200).json({
+        msg: "get popular Blogs",
+    })
+}
+
+const getSimilarBlogs = async (req, res) => {
+    res.status(200).json({
+        msg: "get similar Blogs",
+    })
+}
+
+const getBlogById = async (req, res) => {
+    try {
+        const blogId = req.params.id;
+        const blog = await Blog.findById(blogId).populate('author', 'name email -_id');
+        res.status(200).json(
+            blog
+        )
+    } catch (err) {
+        console.log(err);
+        res.status(500).json({
+            msg: "something went wrong, try again!!!",
+        });
+    }
+}
+
+// draft/publish a blog
+const createBlog = async (req, res) => {
+
+    if (isObjectEmpty(req.body)) {
+        res.status(400).json({
+            msg: "Nothing to create. You provided all fields empty",
+        })
+        return;
+    }
+
+    if (!(req.body.title) || !(req.body.content)) {
```

Comment:
why are checking for empty fields here, can't we use mongoose schema to enforce these checks?